### PR TITLE
build: fuzzy matching library

### PR DESCRIPTION
**Pull request by Misax148**
This pull request contemplates the following:
- Configuration of the Fuzzy Matching library.
- Configuration of Fuzzy Matching dependencies in gradle.
- Establishment of the project base to begin with the functionalities.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+
+### VS Code ###
+.vscode/
+
+### Mac OS ###
+.DS_Store
```
Comment (teran-joseluis) [inline .gitignore]: @Misax148 Can you add .idea/ rule to the file and the empty line too?
So we won't commit every single time in the idea folder when a change is done.

File: src/main/java/org/example/Main.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+        // Press Shift+F10 or click the green arrow button in the gutter to run the code.
+        for (int i = 1; i <= 5; i++) {
+
+            // Press Shift+F9 to start debugging your code. We have set one breakpoint
+            // for you, but you can always add more by pressing Ctrl+F8.
+            System.out.println("i = " + i);
+        }
+    }
+}
```
Comment (teran-joseluis) [inline src/main/java/org/example/Main.java]: @Misax148 Don't forget to add the empty line please.

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    implementation 'org.apache.commons:commons-lang3:3.12.0'
+    implementation 'org.apache.commons:commons-collections4:4.4'
+    implementation 'commons-codec:commons-codec:1.15'
+    implementation 'org.apache.lucene:lucene-core:8.10.1'
+    implementation 'org.apache.lucene:lucene-analyzers-common:8.10.1'
+}
+
+test {
+    useJUnitPlatform()
+}
```
Comment (teran-joseluis) [inline build.gradle]: @Misax148 Don't forget to add the empty line please.

File: src/main/java/org/example/fuzzy_matcher/util/Utils.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                .map(d -> dict.containsKey(d.toLowerCase()) ?
+                        dict.get(d.toLowerCase())
+                        : d)
+                .collect(Collectors.joining(" "));
+    }
+
+    public static boolean isNumeric(String str) {
+        return str.matches(".*\\d.*");
+    }
+}
```
Comment (teran-joseluis) [inline src/main/java/org/example/fuzzy_matcher/util/Utils.java]: @Misax148 Don't forget to add the empty line please.

**Review by sebastian-barra-jalafund — APPROVED** [review]

**Comment by Misax148** [comment]
@teran-joseluis Comments finished!

File: .idea/.gitignore
Comment (teran-joseluis) [inline .idea/.gitignore:1]: @Misax148 Can you avoid to upload or commit .idea folder and files inside of it?
It normally contains sensitive information such as API keys or local paths resulting in a potentially risk to expose this sensitive information.

**Comment by teran-joseluis** [comment]
> @teran-joseluis Comments finished!

@Misax148 I had written one more, please review it.

**Comment by teran-joseluis** [comment]
> @teran-joseluis Comments finished!

> @Misax148 I had written one more comment, please review it.

**Comment by Misax148** [comment]
> > @teran-joseluis Comments finished!
> 
> > @Misax148 I had written one more comment, please review it.

@teran-joseluis Where?

**Review by biancasoliz-uwu — APPROVED** [review]

**Comment by teran-joseluis** [comment]
> > > @teran-joseluis Comments finished!
> > 
> > 
> > > @Misax148 I had written one more comment, please review it.
> 
> @teran-joseluis Where?

@Misax148 I forgot to submit it, but now you can see it. Just saying the comment it is on the .idea/.gitignore path

File: .idea/.gitignore
Comment (Misax148) [reply .idea/.gitignore:1]: Ok, is finished

**Comment by Misax148** [comment]
> > > > @teran-joseluis Comments finished!
> > > 
> > > 
> > > > @Misax148 I had written one more comment, please review it.
> > 
> > 
> > @teran-joseluis Where?
> 
> @Misax148 I forgot to submit it, but now you can see it. Just saying the comment it is on the .idea/.gitignore path

Done

**Review by teran-joseluis — APPROVED** [review]

**Pull request merged**